### PR TITLE
Convolution Reverb Plugin

### PR DESCRIPTION
**Pull request by LiamGallagherBC**
This PR registers the Convolution Reverb plugin in tracktion engine.

Additional small changes to supress MSVC specific warnings.

### CODE REVIEW

File: modules/tracktion_engine/audio_files/tracktion_Thumbnail.cpp [hunk 1]
```diff
@@ -920,7 +920,8 @@ bool TracktionThumbnail::getThumbnailMinMaxValues(int8_t* minValues, int8_t* max
     const juce::ScopedLock sl(lock);
 
     //The length that unity assigns should match the current thumbnail state
-    if (length != numChannels * numberOfThumbSamplesPerChannelToRead)
+    uint32_t curLength = numChannels * numberOfThumbSamplesPerChannelToRead;
+    if (length != curLength)
```
Comment (nlaroche) [inline modules/tracktion_engine/audio_files/tracktion_Thumbnail.cpp]: fixing warnings?

File: modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp [hunk 2]
```diff
@@ -335,7 +335,7 @@ class WaveInputDeviceInstance  : public InputDeviceInstance
             // BEATCONNECT MODIFICATION START
             auto format = getFormatToUse();
             juce::File recordedFile = edit.recordFileRetriever();
-            auto rc = std::make_unique<RecordingContext>(edit.engine, recordedFile, punchIn.inSeconds(), getTargetTracks());
+            auto rc = std::make_unique<RecordingContext>(edit.engine, recordedFile);
```
Comment (nlaroche) [inline modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp]: Why did you change this?

File: modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp [hunk 3]
```diff
@@ -500,11 +500,7 @@ class WaveInputDeviceInstance  : public InputDeviceInstance
     struct RecordingContext
     {
         // BEATCONNECT MODIFICATION START
-        RecordingContext (
-            Engine& p_Engine,
-            const juce::File& p_File, 
-            const double punchIn, 
-            const juce::Array<AudioTrack*>&& p_TrackList)
+        RecordingContext (Engine& p_Engine, const juce::File& p_File)
```
Comment (nlaroche) [inline modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp]: I'm sure its related but I want to understand this change

File: modules/tracktion_engine/audio_files/tracktion_Thumbnail.cpp (hunk 1 above)
Comment (LiamGallagherBC) [reply modules/tracktion_engine/audio_files/tracktion_Thumbnail.cpp]: The discussed changes are no longer part of this PR.

File: modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp (hunk 3 above)
Comment (LiamGallagherBC) [reply modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp]: The discussed changes are no longer part of this PR.

File: modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp (hunk 2 above)
Comment (LiamGallagherBC) [reply modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp]: The discussed changes are no longer part of this PR.
